packages/integration-sdk-runtime/src/execution/__tests__/jobState.test.ts

```diff
@@ -105,8 +136,133 @@ describe('#findEntity', () => {
   });
 
   test('should return "null" if entity not found', async () => {
-    const params = getMockCreateStepJobStateParams();
-    const jobState = createStepJobState(params);
+    const jobState = createTestStepJobStateState();
     expect(await jobState.findEntity('invalid-entity-key')).toEqual(null);
   });
 });
+
+describe('upload callbacks', () => {
+  test('#addEntities should call uploader enqueue on flushed', async () => {
+    const uploadCollector = createInMemoryStepGraphObjectDataUploaderCollector();
+    const jobState = createTestStepJobStateState({
+      graphObjectStore: new FileSystemGraphObjectStore({
+        graphObjectBufferThreshold: 2,
+      }),
+      uploader: uploadCollector.uploader,
+    });
+
+    const e1 = createTestEntity();
+    const e2 = createTestEntity();
+
+    await jobState.addEntities([e1]);
+    await jobState.addEntities([e2]);
+
+    const expectedUploaded: FlushedGraphObjectData[] = [
+      {
+        entities: [e1, e2],
+        relationships: [],
+      },
+    ];
+
+    expect(uploadCollector.graphObjectDataCollection).toEqual(expectedUploaded);
+  });
+
+  test('#addRelationships should call uploader enqueue on flushed', async () => {
+    const uploadCollector = createInMemoryStepGraphObjectDataUploaderCollector();
+    const jobState = createTestStepJobStateState({
+      graphObjectStore: new FileSystemGraphObjectStore({
+        graphObjectBufferThreshold: 2,
+      }),
+      uploader: uploadCollector.uploader,
+    });
+
+    const r1 = createTestRelationship();
+    const r2 = createTestRelationship();
+
+    await jobState.addRelationships([r1]);
+    await jobState.addRelationships([r2]);
+
+    const expectedUploaded: FlushedGraphObjectData[] = [
+      {
+        entities: [],
+        relationships: [r1, r2],
+      },
+    ];
+
+    expect(uploadCollector.graphObjectDataCollection).toEqual(expectedUploaded);
+  });
+
+  test('#flush should call uploader enqueue for entities and relationships', async () => {
+    const uploadCollector = createInMemoryStepGraphObjectDataUploaderCollector();
+    const jobState = createTestStepJobStateState({
+      graphObjectStore: new FileSystemGraphObjectStore({
+        graphObjectBufferThreshold: 5,
+      }),
+      uploader: uploadCollector.uploader,
+    });
+
+    const r1 = createTestRelationship();
+    const e1 = createTestEntity();
+
+    await jobState.addRelationships([r1]);
+    await jobState.addEntities([e1]);
+    await jobState.flush();
+
+    const expectedUploaded: FlushedGraphObjectData[] = [
+      {
+        entities: [e1],
+        relationships: [],
+      },
+      {
+        entities: [],
+        relationships: [r1],
+      },
```

Comment:
Good question. This is because the `FileSystemGraphObjectStore` currently fans out to separate functions to handle the flushing of entities and relationships in parallel.